src/features/staking/StakingModal.tsx

```diff
@@ -205,42 +148,69 @@ const StakingModal: FC<StakingModalInterface> = ({
+      {isRenderManageStake ? (
+        <ManageStake
+          register={register}
+          setValue={setValue}
+          stakeOrUnstake={stakeOrUnstake}
+          setStakeOrUnstake={setStakeOrUnstake}
+          loadingStatus={[statusApprove, statusStake, statusUnstake]}
+        />
       ) : null}
 
-      {statusStaking === "approving" || statusStaking === "staking" ? (
-        <PendingTransaction statusStaking={statusStaking} />
-      ) : null}
+      {/* {isRenderPendingTransaction ? (
+        <PendingTransaction
+          statusStaking={statusStaking}
+          statusUnstake={statusUnstake}
+        />
+      ) : null} */}
 
-      {statusStaking === "approved" || statusStaking === "success" ? (
+      {isRenderApproveSuccess ? (
         <ApproveSuccess
-          statusStaking={statusStaking}
-          setStatusStaking={setStatusStaking}
-          amountApproved={stakingAmount.toString()}
-          amountStaked={stakingAmount.toString()}
+          stakeOrUnstake={stakeOrUnstake}
+          // statusStaking={statusStaking}
+          statusUnstake={statusUnstake}
+          amount={stakingAmount.toString()}
           chainId={chainId}
-          transactionHashApprove={hashApprove?.transactionHash}
-          transactionHashStake={hashStake?.transactionHash}
+          transactionHashStake={transactionReceiptStake?.transactionHash}
+          transactionHashUnstake={transactionReceiptUnstake?.transactionHash}
         />
       ) : null}
 
-      {statusStaking === "failed" ? (
+      {/* {shouldRenderTransactionFailed ? (
         <TransactionFailed
-          setStatusStaking={setStatusStaking}
           chainId={chainId}
-          transactionHash="0x"
+          transactionHashApprove={transactionReceiptUnstake?.transactionHash}
+          transactionHashStake={transactionReceiptApprove?.transactionHash}
+          transactionHashUnstake={transactionReceiptApprove?.transactionHash}
         />
-      ) : null}
-
-      {!isShouldRenderBtn && (
-        <Button
-          className="mb-2 mt-10 w-full rounded-sm bg-accent-blue font-semibold uppercase"
-          onClick={buttonAction}
-        >
-          {buttonText}
-        </Button>
-      )}
+      ) : null} */}
+
+      <Button
+        className={twJoin([
+          "flex flex-row items-center mb-2 mt-10 w-full rounded-sm bg-accent-blue font-semibold uppercase justify-center",
+          `${isButtonDisabled && "opacity-50"}`,
+        ])}
+        onClick={() => {
+          handleButtonActions({
+            stakeOrUnstake,
+            needsApproval,
+            statusStake,
+            statusUnstake,
+            resetStake,
+            resetUnstake,
+            canUnstake,
+            approve,
+            stake,
+            unstake,
+            setValue,
+          });
+        }}
+        disabled={isButtonDisabled}
+      >
+        {isLoadingSpinner && <ImSpinner8 className="animate-spin mr-2 " />}
```

Comment:
For a future PR, build this functionality into the button (i.e. add a `loading` prop), and also disable the disable the button when it's loading.

Reply:
I created a new issue for this for a future PR. Issue: https://github.com/airswap/airswap-voter-rewards/issues/48